### ISSUE TITLE
insertItem - jsGrid does not show up newly inserted row data

### ISSUE THREAD

**Issue by venu-n** [tabalinas/jsgrid#1061]
Form data successfully posted and gets updated in database.  But newly inserted data does not show up in jsGrid, instead blank row is shown.  Checked response separately and the controller returns newly inserted record in JSON format. 
Also tried with "return $.ajax" and "$.ajax", no luck in getting the data displayed in jsGrid, after browser refresh it shows up the newly inserted data.  Not sure what am I missing, please help.

#--- Python  Controller `-----#`
@app.route('/user', methods=['POST'])
def add_user():
    user = mongo.db.user_mgmt
    new_user = []
    user_id = user.insert({'usr_nm':request.form['UserName'],
                           'psswd':request.form['Password'],
                           'isAdmin':request.form['isAdmin']})
    for usr in user.find({'_id':user_id}):
        new_user.append({
            'ID':str(usr['_id']),
            'UserName':usr['usr_nm'],
            'psswd':usr['Password'],
            'isAdmin':usr['isAdmin']})
    return jsonify(new_user)

//--- jsGrid client ----//
insertItem: function (item) {
	return $.ajax({
		type: "POST",
		url: "/user",
		data: item
	});
},

**Comment by tabalinas** [comment]
So you are sure the data has been inserted into the DB?
What's about the response from the server? Did you check dev tools `Network` tab to ensure the response has expected format?
Ensure fields in the response correspond to fields in the grid.

**Comment by venu-n** [comment]
Yes, the data is inserted into the DB and below JSON response is shown in the `Network` tab.
`[
  {
    "ID": "0c83013c-edca-45ea-8272-a2bb2887de39", 
    "Psswd": "sha256$RqW9NeP5$642350bb341885f697fd5fc46aa4380fee9212709acab6526ab55e2118e309da", 
    "UserName": "John", 
    "isAdmin": false
  }
]
`
I cross checked the fields and they correspond to the grid fields.  Still i see a blank row and data shows up only after browser refresh.

**Comment by Adioud** [comment]
Hi,

I had exactly the same problem: new row was added  BUT with empty data (I think that tomvanlier describes another problem).
As often with ennoying problem, answer was very simple at the end...

We have to specify the data type in the $.ajax query.
`dataType: "json"`

Without this, the response is a string, and jsgrid dont find any property to display.

I paste bellow the method of my controller.
Hope this answer doesnt come too late,

Adrien

`insertItem: function(item) {
                item.tool_id = self.id;
                item._token = $('meta[name="csrf-token"]').attr('content');
                return $.ajax({
                    type: "POST",
                    url: "/task",
                    data: item,
                    dataType: "json"
                });
    
            },`

**Comment by DerekLZH** [comment]
Currently I am also developing a similar function, and I also come across with this issue when I try to insert a new row and POST it to the server.

I was using insertItem function which is written inside controller factor:

insertItem: function (item) {
                    var Item = { "ProjectID": item.ProjectID, "ProjectCode": item.ProjectCode, "ProjectName": item.ProjectName, "ProjectStatus": item.ProjectStatus, "Note": item.Note };
                   return $.ajax({
                         type: "POST",
                         url: "<%=Request.Url.LocalPath%>/InsertData",
                         async: false,
                         data: JSON.stringify(Item),
                         contentType: "application/json; charset=utf-8",
                         datatype: "json",
                         success: function (response) {
                            //alert(response);
                            },
                            error: function (xhr, ajaxOptions, thrownError) {
                                alert("Number:" + xhr.status + " Details:" + thrownError);
                            }
                     });

By using the code above, when I insert a new row, the new data for sure is updated to the server, but the grid didn't show up the new data, it shows a blank row instead. Only when I reflash the webpage, it shows up.

Therefore, I don't use the insertItem function. Instead, I use onItemInserted function and it works well.
Below is my code with onItemInserted function:

onItemInserted: function (args) {
                    //Update the new Inserted Data
                    var Item = { "ProjectID": args.item.ProjectID, "ProjectCode": args.item.ProjectCode, "ProjectName": args.item.ProjectName, "ProjectStatus": args.item.ProjectStatus, "Note": args.item.Note };
                    
                    $.ajax({
                        type: "POST",
                        url: "<%=Request.Url.LocalPath%>/InsertData",
                        async: false,
                        data: JSON.stringify(Item),
                        contentType: "application/json; charset=utf-8",
                        datatype: "json",
                        success: function (response) {
                            //alert("Update Successfully!");
                        },
                        error: function (xhr, ajaxOptions, thrownError) {
                            alert("Number:" + xhr.status + " Details:" + thrownError);
                        }
                    });
                    return true;
                },

But honestly, I still don't know why I can achieve my goal by using insertItem..........
